src/fontMetrics.js

```diff
@@ -183,14 +183,28 @@ export type CharacterMetrics = {
     width: number;
 };
 
+export type MetricMap = {
+    [string]: { [string]: number[] }
+}
+let metricMap = {};
+
+/**
+ * This function extends metricMap object with a new metric information
+ */
+export function addFontMetrics(newMetrics: MetricMap) {
+    metricMap = Object.assign(metricMap, newMetrics);
```

Comment:
I'm not sure how `Object.assign` works with nested objects.  Will is merge sub objects or overwrite them?

Reply:
It shallowly merges the source object into destination object so sub objects will be overwritten take a look at [this link](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign#Cloning_an_object).   
It was a intended behaviour to prevent users from partially changing a font metrics. as fonts come as a whole file so their metrics should also come as whole.

Reply:
I'm a little concerned both about the behavior and the naming of this function.  This function can actually override metrics depending on the keys in `newMetrics`.  Also, it could be a partial overwrite.  To make things more predictable, I'd like to suggest changing this to:
```
function setFontMetrics(fontName, metrics) {
   metricMap[fontName] = metrics;
}
```
The problem with this is we have a function called `getFontMetrics` which returns a dictionary of sigma and xi values which are used for positioning things.  The issue is that the names are symmetric, but the values they deal with are completely different which is unexpected.

The thing is, is that `getFontMetrics` actual returns things that look like font metrics.  Font metrics usually refer to values that apply to a font as a whole, e.g. units per em, max ascender, max descender and the like.

Can you change the name of `getFontMetrics` to `getGlobalMetrics`?  Then we can define `setFontMetrics` as described above without there being confusion between the two.